counter/src/main/java/org/infinispan/counter/configuration/CounterConfigurationParser.java

```diff
@@ -0,0 +1,184 @@
+      }
+   }
+
+   @Override
+   public Namespace[] getNamespaces() {
+      return ParseUtils.getNamespaceAnnotations(getClass());
+   }
+
+   private void parseCountersElement(XMLExtendedStreamReader reader, CounterManagerConfigurationBuilder builder)
+         throws XMLStreamException {
+      for (int i = 0; i < reader.getAttributeCount(); i++) {
+         ParseUtils.requireNoNamespaceAttribute(reader, i);
+         String value = replaceProperties(reader.getAttributeValue(i));
+         Attribute attribute = Attribute.forName(reader.getAttributeLocalName(i));
+         switch (attribute) {
+            case NUM_OWNERS:
+               builder.numOwner(Integer.parseInt(value));
+               break;
+            case RELIABILITY:
+               builder.reliability(Reliability.valueOf(value));
+               break;
+            default:
+               throw ParseUtils.unexpectedAttribute(reader, i);
+         }
+      }
+      while (reader.hasNext() && (reader.nextTag() != XMLStreamConstants.END_ELEMENT)) {
+         Element element = Element.forName(reader.getLocalName());
+         switch (element) {
+            case STRONG_COUNTER:
+               parseStrongCounter(reader, builder.addStrongCounter());
+               break;
+            case WEAK_COUNTER:
+               parseWeakCounter(reader, builder.addWeakCounter());
+               break;
+            default:
+               throw ParseUtils.unexpectedElement(reader);
+         }
+      }
+   }
+
+   private void parseWeakCounter(XMLExtendedStreamReader reader, WeakCounterConfigurationBuilder builder)
+         throws XMLStreamException {
+      for (int i = 0; i < reader.getAttributeCount(); i++) {
+         ParseUtils.requireNoNamespaceAttribute(reader, i);
+         String value = replaceProperties(reader.getAttributeValue(i));
+         Attribute attribute = Attribute.forName(reader.getAttributeLocalName(i));
+         switch (attribute) {
+            case CONCURRENCY_LEVEL:
+               builder.concurrencyLevel(Integer.parseInt(value));
+               break;
+            default:
+               parserCommonCounterAttributes(reader, builder, i, attribute, value);
+         }
+      }
+      ParseUtils.requireNoContent(reader);
+   }
+
+   private void parseStrongCounter(XMLExtendedStreamReader reader, StrongCounterConfigurationBuilder builder)
+         throws XMLStreamException {
+      for (int i = 0; i < reader.getAttributeCount(); i++) {
+         ParseUtils.requireNoNamespaceAttribute(reader, i);
+         String value = replaceProperties(reader.getAttributeValue(i));
+         Attribute attribute = Attribute.forName(reader.getAttributeLocalName(i));
+         parserCommonCounterAttributes(reader, builder, i, attribute, value);
+      }
+      while (reader.hasNext() && (reader.nextTag() != XMLStreamConstants.END_ELEMENT)) {
+         Element element = Element.forName(reader.getLocalName());
+         switch (element) {
+            case UPPER_BOUND:
```

Comment:
this was discussed in the last F2F meeting. I don't recall the reasons since we only put the outcome in the minutes :)